symforce/opt/numeric_factor.py

```diff
@@ -71,7 +71,7 @@ def from_file_python(
                 function computes the jacobian with respect to.
             output_dir: The top-level output directory of the linearization function.
             namespace: The namespace of the linearization function.
-            name: The name of the linearization function.
+            name: The name of the linearization function. (Must match the name of the file).
```

Comment:
Instead of just noting it should match the name of the file, we should probably just describe the entire expected structure above the args list:
```
This will load a function `name` from the module located at `output_dir / "python" / "symforce" / namespace / f"{name}.py"`, which matches the directory structure created by `Factor.generate`.
```
```

Reply:
Okay. I changed, it, but I think it might be preferable to instead have this function not take the `output_dir`, but instead take the `function_dir`.

This is because there are really only three scenarios:
1. Loading a function generated by `Codegen.generate_function`, in which case the `function_dir` is part of the returned `GeneratedPaths`.
2. Loading a function generated by `Factor.generate`, which inherits the `GeneratedPaths` from `Codegen.generate_function` in its returned metadata, meaning that's also just as easily available as `output_dir`.
3. A handwritten file, in which case it would be awkward to recreate this directory structure when it doesn't really add much.

Though, I suppose there are some subtletees I'd like to think about first, and since it's not really that important to make the change at the same time, I might just leave that for a subsequent PR.

Reply:
Yeah that all makes sense to me, maybe mildly annoying that you have to look at the result of `sym_factor.generate` in the example, so the common case gets a little more cluttered

Agree we should leave it for a subsequent PR if we want to do it